teemaderegister-be/src/controllers/admin.js

```diff
@@ -1,5 +1,55 @@
+const User = require('../models/user')
+const slug = require('slug')
+const crypto = require('crypto')
+const mail = require('./../utils/mail')
+
+module.exports.createUser = async (req, res) => {
+  const {firstName, lastName, email, role} = req.body
+  const slugs = `${slug(firstName)}-${slug(lastName)}`
+
+  const alreadyUser = await User.findOne({'login.email': email})
+  if (alreadyUser) {
+    return res.status(406).json({message: 'Email already in use!', success: 0})
+  }
+
+  const user = {
+    profile: {
+      firstName: firstName,
+      lastName: lastName,
+      slug: slugs
+    },
+
+    login: {
+      email: email,
+      passwordResetToken: crypto.randomBytes(20).toString('hex'),
+      passwordResetExpires: Date.now() + parseInt(process.env.PASSWORD_RESET_TOKEN_EXPIRES),
+      roles: role
+    }
+  }
+  const passwordCreateURL = `${process.env.SITE_URL}/account/password/${user.login.passwordResetToken}?is-new=1`
+  const newUser = await new User(user).save()
+
+  try {
+    await mail.sendMail({
+      to: email,
+      subject: `Password create`,
+      template: {
+        name: 'createPassword',
+        data: {passwordCreateURL}
+      }
+    })
+  } catch (e) {
```

Comment:
Would actually remove try and catch. If email sending will fail, it will still throw 500